src/services/callAPI.js

```diff
@@ -0,0 +1,48 @@
+// @flow strict
+
+import Promise from 'bluebird'
+
+function handleRequestError(
+  requestInfo: RequestInfo,
+  options: RequestOptions,
+  params: Object,
+  retryCount: number,
+): Promise<any> {
+  return Promise
+    .delay((2 ** ((retryCount > 6) ? 6 : retryCount)) * 1000)
+    // eslint-disable-next-line no-use-before-define
+    .then(() => callAPI(
+      requestInfo,
+      options,
+      params,
+      retryCount + 1,
+    ))
+}
+
+export function callAPI(
+  requestInfo: RequestInfo,
+  options: RequestOptions,
+  params?: ?Object,
+  retryCount?: number = 0,
+): Promise<any> {
+  return fetch(requestInfo, {
+    ...options,
+    body: params ? JSON.stringify(params) : null,
+  }).catch(() => handleRequestError(
+    requestInfo,
+    options,
+    params,
+    retryCount,
+  )).then((response: Response): Promise<any> => {
+    if (response.ok) {
+      return response.json()
+    }
+
```

Comment:
Let's also send exception to GA with future migration to Sentry. Docs on GA exceptions handling are here: https://developers.google.com/analytics/devguides/collection/analyticsjs/exceptions